security-admin/src/main/java/org/apache/ranger/rest/PublicAPIs.java

```diff
@@ -415,4 +418,44 @@ public RangerManagementConfig getManagementConfiguration(@Context HttpServletReq
 		return rangerManagementConfig;
 	}
 
+	@GET
+	@Path("/admin/ok")
+	public String checkRangerHealth (@Context HttpServletRequest request) {
+
+		if (logger.isDebugEnabled()) {
+			logger.debug("==> PublicAPIs.checkRangerHealth(): ");
+		}
+		HashMap<String, String> responseMap = new HashMap<>();
+		JSONUtil jsonUtil = new JSONUtil();
+
+		// Default Response Parameters
+		String serviceStatus = "partially running";
+		String dbStatus = "not running";
+		String version = RangerVersionInfo.getVersion();
+		boolean isSuccess = false;
+
+		// Get Database Version
+		String dbVersion = daoMgr.getXXDataHist().getDBVersion();
+
+		if (!dbVersion.equalsIgnoreCase(BaseDao.NOT_AVAILABLE)) {
```

Comment:
a small code nitpick,  I suggest flipping the condition to guard against any NPE

if (!BaseDao.NOT_AVAILABLE.equalsIgnoreCase(dbVersion)) 

the rest looks good please self-merge after the change.